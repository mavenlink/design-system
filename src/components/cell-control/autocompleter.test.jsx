import React, { createRef } from 'react';
import { render as _render } from '@testing-library/react';
import Autocompleter from './autocompleter.jsx';

const render = (ui, options = { labelledBy: 'labelled-by' }) => (
  _render(ui, {
    ...options,
    wrapper: props => (
      <table>
        <thead>
          <tr>
            <th id={options.labelledBy}>Column Header</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            {props.children}
          </tr>
        </tbody>
      </table>
    ),
  })
);

describe('Autocompleter cell control', () => {
  const requiredProps = {
    id: 'test-id',
    labelledBy: 'labelled-by',
    name: 'test-name',
  };

  it('has defaults', () => {
    const ref = createRef();
    render(<Autocompleter {...requiredProps} ref={ref} />);
    expect(document.body).toMatchSnapshot();
    expect(document.body).toBe(document.activeElement);
    expect(ref.current).toMatchSnapshot();
  });

  describe('className API', () => {
    it('can be set', () => {
      render(<Autocompleter {...requiredProps} className="unique-class-name" />);
      expect(document.body).toMatchSnapshot();
    });
  });
});
