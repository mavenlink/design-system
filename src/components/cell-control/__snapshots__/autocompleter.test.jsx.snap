// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocompleter cell control className API can be set 1`] = `
<body>
  <div>
    <table>
      <thead>
        <tr>
          <th
            id="labelled-by"
          >
            Column Header
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            aria-labelledby="labelled-by"
            class="unique-class-name"
            role="gridcell"
          >
            <div
              class="container"
              style="position: relative;"
            >
              <input
                aria-autocomplete="none"
                aria-controls="test-id-single-choice-listbox"
                aria-describedby="test-id-tooltip test-idHint"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="input"
                id="test-id"
                name="test-name"
                role="combobox"
                style="--numIcon: 3;"
                type="text"
                value=""
              />
              <div
                class="icon-container"
              >
                <svg
                  class="input-icon"
                  height="6"
                  role="img"
                  width="10"
                >
                  <title>
                    Open choices listbox
                  </title>
                  <use
                    xlink:href="#caret-down.svg"
                  />
                </svg>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;

exports[`Autocompleter cell control has defaults 1`] = `
<body>
  <div>
    <table>
      <thead>
        <tr>
          <th
            id="labelled-by"
          >
            Column Header
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            aria-labelledby="labelled-by"
            class="container"
            role="gridcell"
          >
            <div
              class="container"
              style="position: relative;"
            >
              <input
                aria-autocomplete="none"
                aria-controls="test-id-single-choice-listbox"
                aria-describedby="test-id-tooltip test-idHint"
                aria-expanded="false"
                aria-haspopup="listbox"
                autocomplete="off"
                class="input"
                id="test-id"
                name="test-name"
                role="combobox"
                style="--numIcon: 3;"
                type="text"
                value=""
              />
              <div
                class="icon-container"
              >
                <svg
                  class="input-icon"
                  height="6"
                  role="img"
                  width="10"
                >
                  <title>
                    Open choices listbox
                  </title>
                  <use
                    xlink:href="#caret-down.svg"
                  />
                </svg>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</body>
`;

exports[`Autocompleter cell control has defaults 2`] = `
Object {
  "dirty": false,
  "name": "test-name",
  "value": undefined,
}
`;
